### PR TITLE
Add personality uses

### PR DESCRIPTION
**Pull request by tuvus**
This PR adds in some implementation of the personality values.
The changes are centered around MotivationToAttackAutomation and DiplomacyAutomation.
I also changed MotivationToAttackAutomation to use floats instead of ints.

As always, this is not the last time I will tweak this area of code. I'll probably start working on another one once this gets merged.

### CODE REVIEW

File: core/src/com/unciv/logic/automation/civilization/MotivationToAttackAutomation.kt [hunk 1]
```diff
@@ -55,70 +61,74 @@ object MotivationToAttackAutomation {
...
         if (diplomacyManager.resourcesFromTrade().any { it.amount > 0 })
-            modifiers.add(Pair("Receiving trade resources", -5))
+            modifiers.add(Pair("Receiving trade resources", -8f * personality.modifierFocus(PersonalityValue.Commerce, .5f) * personality.modifierFocus(PersonalityValue.Loyal, .2f)))
 
         // If their cities don't have any nearby cities that are also targets to us and it doesn't include their capital
         // Then there cities are likely isolated and a good target.
         if (targetCiv.getCapital(true) !in targetCities
                 && targetCities.all { theirCity -> !theirCity.neighboringCities.any { it !in targetCities } }) {
-            modifiers.add(Pair("Isolated city", 15))
+            modifiers.add(Pair("Isolated city", 10f * personality.modifierFocus(PersonalityValue.Aggressive, .8f)))
```
Comment (tuvus) [inline core/src/com/unciv/logic/automation/civilization/MotivationToAttackAutomation.kt:114]: Aggressive civs are move likely to make use of the isolated city modifier.

File: core/src/com/unciv/logic/automation/civilization/MotivationToAttackAutomation.kt [hunk 2]
```diff
@@ -145,137 +155,140 @@ object MotivationToAttackAutomation {
...
-            scoreRatio > .25f -> -5
-            else -> -10
+            scoreRatio > 2f -> 15f
+            scoreRatio > 1.5f -> 10f
+            scoreRatio > 1.25f -> 5f
+            scoreRatio > 1f -> 2f
+            scoreRatio > .8f -> 0f
+            scoreRatio > .5f -> -2f
+            scoreRatio > .25f -> -5f
+            else -> -10f
```
Comment (tuvus) [inline core/src/com/unciv/logic/automation/civilization/MotivationToAttackAutomation.kt:254]: I slightly reduced the motivation from higher score civilizations.

File: core/src/com/unciv/logic/automation/civilization/MotivationToAttackAutomation.kt [hunk 3]
```diff
@@ -145,137 +155,140 @@ object MotivationToAttackAutomation {
...
+            combatStrengthRatio > 3f -> 12f
+            combatStrengthRatio > 2f -> 10f
+            combatStrengthRatio > 1.8f -> 8f
+            combatStrengthRatio > 1.6f -> 6f
+            combatStrengthRatio > 1.4f -> 4f
+            combatStrengthRatio > 1.2f -> 2f
+            combatStrengthRatio > .8f -> 0f
+            combatStrengthRatio > .6f -> -5f
+            combatStrengthRatio > .4f -> -15f
+            else -> -20f
```
Comment (tuvus) [inline core/src/com/unciv/logic/automation/civilization/MotivationToAttackAutomation.kt:289]: I reduced the max motivation from strength here.

File: core/src/com/unciv/logic/trade/Trade.kt [hunk 4]
```diff
@@ -57,9 +58,9 @@ class TradeRequest : IsPartOfGameInfoSerialization {
         // the numbers of the flags (20,5) are the amount of turns to wait until offering again
         if (trade.ourOffers.all { it.type == TradeType.Luxury_Resource }
             && trade.theirOffers.all { it.type == TradeType.Luxury_Resource })
-            requestingCivDiploManager.setFlag(DiplomacyFlags.DeclinedLuxExchange,5)
+            requestingCivDiploManager.setFlag(DiplomacyFlags.DeclinedLuxExchange,5 - (requestingCivInfo.getPersonality()[PersonalityValue.Commerce] / 2).toInt())
         if (trade.ourOffers.any { it.name == Constants.researchAgreement })
-            requestingCivDiploManager.setFlag(DiplomacyFlags.DeclinedResearchAgreement,10)
+            requestingCivDiploManager.setFlag(DiplomacyFlags.DeclinedResearchAgreement,15 - requestingCivInfo.getPersonality()[PersonalityValue.Science].toInt())
```
Comment (tuvus) [inline core/src/com/unciv/logic/trade/Trade.kt:63]: Personality can affect the rate at which the AI trades and asks for research agreements

**Comment by tuvus** [comment]
In some future PR I plan on adding logic to check for competitors. One general goal of the AI is to have smaller nations team up on slightly bigger nations to make it more competitive. It seems like I need to implement this behavior a little more directly.

File: core/src/com/unciv/logic/automation/Automation.kt [hunk 5]
```diff
@@ -233,8 +233,8 @@ object Automation {
         if (civInfo.gameInfo.turns > 120 * speed.barbarianModifier * multiplier)
             multiplier /= 2
 
-        // If we have a lot of, or no cities we are not afraid
-        if (civInfo.cities.isEmpty() || civInfo.cities.size >= 4 * multiplier)
+        // If we have no cities or a lot of units we are not affraid
```
[the remark this answers is not in the document]
Comment (tuvus) [reply core/src/com/unciv/logic/automation/Automation.kt:236]: What amazes me is how you spot these typos.

**Review by yairm210 — APPROVED** [review]

**Pull request merged**